### PR TITLE
feat: Consume release date from vestings

### LINKED ISSUE #1505: Consume transactions from Vesting contracts for Financial Report Updates

### PR DESCRIPTION
**Pull request by ncomerci**
**Screenshots**
image

**Related Issue**
Closes #1505 

- [ ]  **Functionality validation**
    - [ ]  Feature working as described in the related issue
    - [ ]  New environment variables have been set
    - [ ]  New API services methods have tests
    - [ ]  Code errors are being logged in the code health service
- [ ]  **Product validation**
    - [ ]  UI changes look the same as in Figma
    - [ ]  Usage events are being sent to the product analytics service
    - [ ]  New common components have been documented in Storybook
    - [ ]  Preview link if it’s a customer facing issue
- [ ] **Documentation**
    - [ ] README updated
    - [ ] New API changes are documented
- [ ] **Accessibility**
    - [ ] UI changes are accessible

### CODE REVIEW

**Review by 1emu — APPROVED** [review]

File: src/components/Updates/FinancialSection.tsx [hunk 1]
```diff
@@ -210,6 +210,10 @@ function FinancialSection({
     })
   }, [csvInputField, csvInputHandler, readString])
 
+  const lastUpdate = publicUpdates?.filter(
```
[the remark this answers is not in the document]
Comment (andyesp) [reply src/components/Updates/FinancialSection.tsx:213]: Out of schedule updates will always occur after the scheduled ones. So we could assume the first out of schedule update is the last one the user posted.

**Review by andyesp — APPROVED** [review]

File: src/components/Updates/FinancialCardsSection.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  const undisclosedFunds = disclosedFunds <= releasedFundsValue ? releasedFundsValue - disclosedFunds : 0
+  const { formatNumber } = useIntl()
+  return (
+    <div className="FinancialCardsSection__CardsContainer">
+      <FinancialCard
+        type="income"
+        title={t('page.proposal_update.funds_released_label')}
+        value={releasedFundsValue}
+        subtitle={
+          lastRelease
```
Comment (1emu) [inline src/components/Updates/FinancialCardsSection.tsx:32]: Similar question as before: is this the last release or is it the latest? :)

File: src/entities/Updates/utils.ts [hunk 3]
```diff
@@ -90,38 +97,42 @@ export function getUpdateNumber(publicUpdates?: UpdateAttributes[] | null, updat
...
-  const monthsSinceLastUpdate = Math.ceil(Time(now).diff(due_date, 'month', true))
+export function getReleases(vestings: VestingInfo[]) {
+  return vestings
+    .flatMap(({ logs }) => logs)
+    .filter(({ topic }) => RELEASE_TOPICS.has(topic))
+    .sort((a, b) => new Date(b.timestamp).getTime() - new Date(a.timestamp).getTime())
+}
 
-  return vestedPerMonth * monthsSinceLastUpdate
+export function getLatestUpdate(publicUpdates: UpdateAttributes[], beforeDate?: Date): UpdateAttributes | undefined {
```
Comment (1emu) [inline src/entities/Updates/utils.ts:126]: I think you can write this like

```typescript
export function getLatestUpdate(publicUpdates: UpdateAttributes[], beforeDate?: Date): UpdateAttributes | undefined {
  if (!publicUpdates?.length) return undefined;

  return publicUpdates
    .filter(update => update.status === UpdateStatus.Done || update.status === UpdateStatus.Late)
    .reduce((latest, current) => {
      if (!latest || !latest.completion_date || (beforeDate && Time(latest.completion_date).isAfter(beforeDate))) {
        return current;
      }
      return Time(latest.completion_date).isAfter(current.completion_date) ? latest : current;
    }, undefined);
}
```

**Review by 1emu — APPROVED** [review]

File: src/entities/Updates/utils.ts [hunk 4]
```diff
@@ -90,38 +97,41 @@ export function getUpdateNumber(publicUpdates?: UpdateAttributes[] | null, updat
...
-
-  if (Time(vestingStartDate).isAfter(now)) {
-    return 0
-  }
-
-  const lastUpdate = updates?.filter(
-    (update) => update.status === UpdateStatus.Done || update.status === UpdateStatus.Late
-  )?.[0]
+export function getFundsReleasedSinceLastestUpdate(
+  lastestUpdate: Omit<UpdateAttributes, 'id' | 'proposal_id'> | undefined,
```
Comment (andyesp) [inline src/entities/Updates/utils.ts:101]: Typo `latestUpdate`

File: src/entities/Updates/utils.ts [hunk 5]
```diff
@@ -90,38 +97,41 @@ export function getUpdateNumber(publicUpdates?: UpdateAttributes[] | null, updat
...
+  if (!lastestUpdate) {
+    return { value: sum(releases.map(({ amount }) => amount || 0)), txAmount: releases.length }
   }
 
-  const { due_date } = lastUpdate
+  const { completion_date } = lastestUpdate
 
-  if (!due_date) {
-    return 0
+  const releasesSinceLastestUpdate = releases.filter(({ timestamp }) => Time(timestamp).isAfter(completion_date))
```
Comment (andyesp) [inline src/entities/Updates/utils.ts:112]: Typo `releasesSinceLatestUpdate`

File: src/pages/update.tsx [hunk 6]
```diff
@@ -47,6 +48,8 @@ export default function UpdateDetail() {
   const index = getUpdateNumber(publicUpdates, updateId)
   const proposalHref = locations.proposal(update.proposal_id)
 
+  const lastestUpdate = getLatestUpdate(publicUpdates || [], Time(update.due_date).toDate())
```
Comment (andyesp) [inline src/pages/update.tsx:51]: Again `latestUpdate`

**Review by andyesp — APPROVED** [review]

**Review by 1emu — APPROVED** [review]

**Pull request merged**